app/src/org/commcare/activities/PostRequestActivity.java

```diff
@@ -0,0 +1,246 @@
+    private static final String IN_ERROR_STATE_KEY = "in-error-state-key";
+    private static final String ERROR_MESSAGE_KEY = "error-message-key";
+
+    public static final String URL_KEY = "url-key";
+    public static final String PARAMS_KEY = "params-key";
+
+    @UiElement(value = R.id.request_button, locale = "post.request.button")
+    private Button retryButton;
+
+    @UiElement(value = R.id.error_message)
+    private TextView errorMessageBox;
+
+    private URL url;
+    private HashMap<String, String> params;
+    private String errorMessage;
+    private boolean hasTaskLaunched;
+    private boolean inErrorState;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        loadStateFromSavedInstance(savedInstanceState);
+        loadStateFromIntent(getIntent());
+
+        setupUI();
+
+        if (!inErrorState) {
+            makePostRequest();
+        }
+    }
+
+    private void loadStateFromSavedInstance(Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            hasTaskLaunched = savedInstanceState.getBoolean(TASK_LAUNCHED_KEY);
+            inErrorState = savedInstanceState.getBoolean(IN_ERROR_STATE_KEY);
+            errorMessage = savedInstanceState.getString(ERROR_MESSAGE_KEY);
+        }
+    }
+
+    private void loadStateFromIntent(Intent intent) {
+        if (intent.hasExtra(URL_KEY) && intent.hasExtra(PARAMS_KEY)) {
+            String urlString = intent.getStringExtra(URL_KEY);
+            try {
+                url = new URL(urlString);
+            } catch (MalformedURLException e) {
+                enterErrorState(Localization.get("post.malformed.url", urlString));
+            }
+            Object o = intent.getSerializableExtra(PARAMS_KEY);
+            params = (HashMap<String, String>)o;
+        } else {
+            enterErrorState(Localization.get("post.generic.error"));
+        }
+    }
+
+    private void setupUI() {
+        if (inErrorState) {
+            enterErrorState();
+        } else {
+            retryButton.setVisibility(View.GONE);
+        }
+
+        retryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                retryPost();
+            }
+        });
+    }
+
+    private void performSync() {
+        (new FormAndDataSyncer()).syncDataForLoggedInUser(this, false, false);
+    }
+
+    private void makePostRequest() {
+        if (!hasTaskLaunched && !inErrorState) {
+            SimpleHttpTask postTask;
+            try {
+                postTask = new SimpleHttpTask(this, url, params, true);
+            } catch (ModernHttpRequester.PlainTextPasswordException e) {
+                enterErrorState(Localization.get("post.not.using.https", url.toString()));
+                return;
+            }
+            postTask.connect((CommCareTaskConnector)this);
```

Comment:
Is it because the implementation is indirect (`PostRequestActivity` implements `ConnectorWithHttpResponseProcessor` which implements `CommCareTaskConnector` ?)